### PR TITLE
Kickoff api

### PR DESCRIPTION
**Pull request by ivoguzman23**
## Summary

Registration endpoint

I will change the routes in the next pr, but now you can made a POST to http://localhost:3000/signup
Yet I dont configure the new params, that is why first_name and last_name are null.
I have to log the user after registration but i am still working on it

## Screenshots

image

## TRELLO Card

https://trello.com/c/6RLF1wKR/41-implementar-registraci%C3%B3n-de-usuarios

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -25,3 +25,6 @@
 
 # Ignore master key for decrypting credentials and more.
 /config/master.key
+
+# Ignore .env files
+*.env
```
Comment (GabrielPiccini) [inline .gitignore]: Here you should write only `.env` because this file is in the root of your proyect so the asterisk is unnesesary

File: .env [hunk 2]
```diff
@@ -0,0 +1 @@
+DEVISE_JWT_SECRET_KEY=44a238d54045d8fe4a9a9cf3fd5a530ebb77d6122aff436d212917c65919ccda9e106ee1cbcd335232924657288c406433d80d3ed4aa3c8d675664d115421e00
```
Comment (GabrielPiccini) [inline .env:1]: This file should not be here

File: app/controllers/users/sessions_controller.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+  def respond_with(resource, _opts = {})
+    render json: resource
+  end
+
+  def respond_to_on_destroy
+    head :no_content
+  end
+
+  # GET /resource/sign_in
```
Comment (GabrielPiccini) [inline app/controllers/users/sessions_controller.rb]: You should delete unnesesary comments

File: app/controllers/users/registrations_controller.rb [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+  respond_to :json
+
+  def create
+    build_resource(sign_up_params)
+
+    resource.save
+    render_resource(resource)
+  end
+  # GET /resource/sign_up
```
Comment (GabrielPiccini) [inline app/controllers/users/registrations_controller.rb]: You should delete unnesesary comments

File: app/controllers/users/confirmations_controller.rb [hunk 5]
```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+class Users::ConfirmationsController < Devise::ConfirmationsController
+  # GET /resource/confirmation/new
+  # def new
```
Comment (GabrielPiccini) [inline app/controllers/users/confirmations_controller.rb]: You should delete unnesesary comments

File: app/controllers/registrations_controller.rb [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+  respond_to :json
+
+  def create
+    build_resource(sign_up_params)
+
+    resource.save
+    render_resource(resource)
+  end
+  # GET /resource/sign_up
+  # def new
```
Comment (GabrielPiccini) [inline app/controllers/registrations_controller.rb]: You should delete unnesesary comments

File: app/controllers/sessions_controller.rb [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+  def respond_with(resource, _opts = {})
+    render json: resource
+  end
+
+  def respond_to_on_destroy
+    head :no_content
+  end
+
+  # GET /resource/sign_in
+  # def new
```
Comment (GabrielPiccini) [inline app/controllers/sessions_controller.rb]: You should delete unnesesary comments

File: app/controllers/users/omniauth_callbacks_controller.rb [hunk 8]
```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+class Users::OmniauthCallbacksController < Devise::OmniauthCallbacksController
+  # You should configure your model like this:
+  # devise :omniauthable, omniauth_providers: [:twitter]
```
Comment (GabrielPiccini) [inline app/controllers/users/omniauth_callbacks_controller.rb]: You should delete unnesesary comments

File: app/controllers/users/passwords_controller.rb [hunk 9]
```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+class Users::PasswordsController < Devise::PasswordsController
+  # GET /resource/password/new
+  # def new
+  #   super
```
Comment (GabrielPiccini) [inline app/controllers/users/passwords_controller.rb]: You should delete unnesesary comments

File: app/controllers/users/unlocks_controller.rb [hunk 10]
```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+class Users::UnlocksController < Devise::UnlocksController
+  # GET /resource/unlock/new
```
Comment (GabrielPiccini) [inline app/controllers/users/unlocks_controller.rb]: You should delete unnesesary comments

**Review by GabrielPiccini — CHANGES_REQUESTED** [review]
Changes

File: db/migrate/20210616124819_add_first_name_and_last_name.rb [hunk 11]
```diff
@@ -0,0 +1,4 @@
+class AddFirstNameAndLastName < ActiveRecord::Migration[5.2]
+  def change
+  end
+end
```
Comment (lucasenriquez) [inline db/migrate/20210616124819_add_first_name_and_last_name.rb]: What is this file for?

File: db/migrate/blacklist_migration.rb [hunk 12]
```diff
@@ -0,0 +1,9 @@
+class CreateJwtBlacklist < ActiveRecord::Migration[5.0]
+    def change
+      create_table :jwt_blacklist do |t|
+        t.string :jti, null: false
+      end
+      add_index :jwt_blacklist, :jti
+    end
+  end
+
```
Comment (lucasenriquez) [inline db/migrate/blacklist_migration.rb]: Add a newline at the end of the file

File: test/fixtures/users.yml [hunk 13]
```diff
@@ -0,0 +1,11 @@
...
+
+# This model initially had no columns defined. If you add columns to the
+# model remove the '{}' from the fixture names and add the columns immediately
+# below each fixture, per the syntax in the comments below
+#
+one: {}
+# column: value
+#
+two: {}
+# column: value
```
Comment (lucasenriquez) [inline test/fixtures/users.yml]: What is this file for?

File: test/models/user_test.rb [hunk 14]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
+
+class UserTest < ActiveSupport::TestCase
+  # test "the truth" do
+  #   assert true
+  # end
+end
```
Comment (lucasenriquez) [inline test/models/user_test.rb]: What is this file for?

**Review by lucasenriquez — CHANGES_REQUESTED** [review]

**Comment by ivoguzman23** [comment]
.

**Pull request closed without merging**